src/main/java/uk/gov/pay/api/resources/LedgerOrConnectorStrategyTemplate.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.pay.api.resources;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public abstract class LedgerOrConnectorStrategyTemplate<T> {
+
+    private static final String HEADER_NAME = "X-Ledger";
+    private static final Logger logger = LoggerFactory.getLogger(LedgerOrConnectorStrategyTemplate.class);
+    private String strategy;
+    private List<String> VALID_STRATEGIES = ImmutableList.of("ledger-only", "future-behaviour");
+    
+    public LedgerOrConnectorStrategyTemplate(String strategy) {
+        this.strategy = strategy;
+    }
+
+    private void validate() {
+        if(strategy != null && !StringUtils.isEmpty(strategy) && !VALID_STRATEGIES.contains(strategy)) {
```

Comment:
!StringUtils.isBlank(strategy) could be used instead of checking for null. Also, isEmpty returns false if there is whitespace only in the String.

Reply:
Sounds good to me - will update.